internal/server/routes.go

```diff
@@ -56,7 +57,14 @@ func (s *Server) GenerateRoutes() Routes {
 
 	get(a, authn, "/api/users", a.ListUsers)
 	post(a, authn, "/api/users", a.CreateUser)
-	get(a, authn, "/api/users/:id", a.GetUser)
+	// getuser needs its own route declaration because it may call database write operations on sync
+	getUserRoute := route[api.GetUserRequest, *api.User]{
```

Comment:
The pattern we've been using for other routes is to move this `getUserRoute` variable next to where the handler (`API.GetUser`) is defined. That way this list of routes stays concise, and it's easy to see which route settings are applied to the handler when looking at the handler.

I don't think hiding those settings in functions is a great option here. All we're doing is setting values on a struct. A variable for common settings could work well (we have `defaultRouteSettingsGet` in #2006), but I doubt we want a separate one for each permutation. I think something in the middle works well: a shared variable for the most common cases, but any route that does something unique should just define the settings for that one route.

I think it's pretty important to avoid write operations in `GET` requests (outside of a few isolated places in the "middleware"). I'm hoping we can change where this update happens to avoid the need for a custom route setting here.

Reply:
Removed this update to the get request